.github/workflows/pre-merge-checks.yml

```diff
@@ -131,7 +157,7 @@ jobs:
             !delivery-receipts-integration-tests/build/reports/
       - name: Upload Unit Test Reports
         uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882 # v4.4.3
-        if: failure()
+        if: failure() && needs.check-changed-files.outputs.java_changed == 'true'
```

Comment:
Why have we decided to do this differently to the previous steps?  Should we have `if: needs.check-changed-files.outputs.java_changed == 'true'` as the first line.

Reply:
so here, specifically the `failure()` is the primary condition we're checking.

So this is if the test failed *and* if Java files changed.

See also: https://github.com/govuk-one-login/authentication-api/blob/7bcca5494d3f4c99a1c24a1ef05e748fbf413fc4/.github/workflows/pre-merge-checks.yml#L248 https://github.com/govuk-one-login/authentication-api/blob/7bcca5494d3f4c99a1c24a1ef05e748fbf413fc4/.github/workflows/pre-merge-checks.yml#L325 https://github.com/govuk-one-login/authentication-api/blob/7bcca5494d3f4c99a1c24a1ef05e748fbf413fc4/.github/workflows/pre-merge-checks.yml#L348 https://github.com/govuk-one-login/authentication-api/blob/7bcca5494d3f4c99a1c24a1ef05e748fbf413fc4/.github/workflows/pre-merge-checks.yml#L424 for similar examples

Reply:
My point was that the other steps had the have files changed check as the first thing they look at so it would be consistent if all steps did this.  If no files have changed isn't it true that no tests will have run and so there's no point looking at failure()?  Its a minor point so will resolve conversation.